presto-sqlserver/src/main/java/io/prestosql/plugin/sqlserver/SqlServerClient.java

```diff
@@ -233,6 +240,32 @@ public boolean isLimitGuaranteed(ConnectorSession session)
         return true;
     }
 
+    @Override
+    protected String createTableSql(RemoteTableName remoteTableName, List<String> columns, ConnectorTableMetadata tableMetadata)
+    {
+        return format("CREATE TABLE %s (%s) WITH (DATA_COMPRESSION = %s)", quoted(remoteTableName), join(", ", columns), getDataCompression(tableMetadata.getProperties()));
+    }
+
+    @Override
+    public Map<String, Object> getTableProperties(JdbcIdentity identity, JdbcTableHandle tableHandle)
+    {
+        ImmutableMap.Builder<String, Object> builder = ImmutableMap.builder();
+
+        try (Connection connection = connectionFactory.openConnection(identity);
+                PreparedStatement statement = connection.prepareStatement("SELECT data_compression_desc FROM sys.partitions p INNER JOIN sys.tables t ON p.object_id = t.object_id WHERE p.partition_number = 1 AND t.name = ?")) {
```

Comment:
As you explained to me offline, the SQL Server treats table with single partition as an unpartitioned table.
Is there a guarantee that the only partition will have `partition_number=1`?
Also, in case of a partitioned table, we should _not_ return compression info from the first partition.

Reply:
From their documentation it looks like they set `partition_number` as 1 for non-partitioned tables. 
Ref : https://docs.microsoft.com/en-us/sql/relational-databases/system-catalog-views/sys-partitions-transact-sql?view=sql-server-ver15

Reply:
This answers the first question.

Reply:
From their docs it looks like 1 is used only for non-partitioned tables